### PR TITLE
Connect React Admin to the react-router app

### PR DESCRIPTION
**Pull request by dtrucs**
ref #8

### CODE REVIEW

File: src/views/Main/MenuDropdown/index.js [hunk 1]
```diff
@@ -4,6 +4,14 @@ import { connectAppProvider } from '../../../components/AppProvider';
 import { getComponentsByEnabledModules } from '../../../services/modules';
 import { MenuDropdown } from './MenuDropdown';
 
-export default connectAppProvider(
-  ({ env: { enabled_modules: modules } }) => ({ modules: getComponentsByEnabledModules(modules) }),
-)(withNamespaces()(withRouter(MenuDropdown)));
+export default withRouter(
+  connectAppProvider(
+    ({
```
Comment (mabhub) [inline src/views/Main/MenuDropdown/index.js]: Please put `connectAppProvider()` parameters in a variable to improve readability.

Maybe something like: 
```js
const findMeABetterName = ({ env: { enabled_modules: modules } }) => ({ modules: getComponentsByEnabledModules(modules) });

export default withRouter(
  connectAppProvider(findMeABetterName)(
    withNamespaces()(MenuDropdown),
  ),
);
```

File: src/modules/CRUD/views/Map/index.js [hunk 2]
```diff
@@ -6,29 +6,31 @@ import { connectCRUDProvider } from '../../services/CRUDProvider';
...
+      'layersList',
+      'setMap',
+      'map',
+      'resizingMap',
+      'mapIsResizing',
+      'getFeaturesList',
+      'featuresList',
+      'feature',
+    )(
+      withNamespaces()(
```
Comment (mabhub) [inline src/modules/CRUD/views/Map/index.js]: Not really sure that these line breaks improve readability..

```js
      withNamespaces()(Map),
```

File: src/modules/User/User.js [hunk 3]
```diff
@@ -11,19 +12,20 @@ import userViews from './views';
...
     <Admin
       appLayout={RALayout}
-      {...providers}
       locale={`${locale}`.substr(0, 2)}
+      history={history}
+      {...providers}
     >
       <Resource
-        name="user"
+        name={`${config.path.substring(1)}/user`}
```
[the remark this answers is not in the document]
Comment (dtrucs) [reply src/modules/User/User.js]: With the last change, it calls a function to get the good path

File: src/services/react-admin/enhanceDataProvider.js [hunk 4]
```diff
@@ -2,7 +2,9 @@ import Api from '@terralego/core/modules/Api';
 import { WMTS } from '../../modules/DataSource/DataSource';
 
 const enhanceDataProvider = mainDataProvider => async (...args) => {
-  const [type, resource, params] = args;
+  const [type, resourceWithBasePath, params] = args;
+
+  const resource = resourceWithBasePath.replace(/((?:.*)\/)/, '');
```
[the remark this answers is not in the document]
Comment (dtrucs) [reply src/services/react-admin/enhanceDataProvider.js]: pure function 👍

File: src/views/Main/MenuDropdown/MenuDropdown.js [hunk 5]
```diff
@@ -8,7 +10,7 @@ import {
   Position,
 } from '@blueprintjs/core';
 
-export const MenuDropdown = ({ t, modules = [], history: { push } }) => {
+export const MenuDropdown = ({ t, modules = [] }) => {
```
[the remark this answers is not in the document]
Comment (dtrucs) [reply src/views/Main/MenuDropdown/MenuDropdown.js]: Done

File: src/views/Main/MenuDropdown/index.js (hunk 1 above)
Comment (dtrucs) [reply src/views/Main/MenuDropdown/index.js]: Done

File: src/modules/CRUD/views/Map/index.js (hunk 2 above)
Comment (dtrucs) [reply src/modules/CRUD/views/Map/index.js]: Done

File: src/services/react-admin/utils.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Remove the base path if it exists
+ * example:
+ * 'foo/bar' returns 'bar'
+ * 'foobar' returns 'foobar'
+ * @param {string} string
+ * @returns {string}
+ */
+export function getResourceWithoutBasePath (string) {
```
Comment (mabhub) [inline src/services/react-admin/utils.js]: This should be an *Arrow function*.

Comment (mabhub) [reply src/services/react-admin/utils.js]: I close this in favor of the newly created issue.

**Pull request merged**